libs/state/src/lib/medical-illustration/medical-illustration.state.ts

```diff
@@ -0,0 +1,38 @@
+import { inject, Injectable } from '@angular/core';
+import { Action, State, StateContext } from '@ngxs/store';
+import { parse, ParseResult } from 'papaparse';
+
+import { SetActiveNode, SetMapping, SetUri } from './medical-illustration.actions';
+import { MedicalIllustrationModel } from './medical-illustration.model';
+
+export type MedicalIllustrationContext = StateContext<MedicalIllustrationModel>;
+
+@State<MedicalIllustrationModel>({
+  name: 'medicalIllustration',
+  defaults: {},
+})
+@Injectable()
+export class MedicalIllustrationState {
+  @Action(SetUri)
+  setUri({ setState }: MedicalIllustrationContext, { url }: SetUri) {
+    setState({ url });
+  }
+
+  @Action(SetActiveNode)
+  setActiveNode({ patchState }: MedicalIllustrationContext, { node }: SetActiveNode) {
+    patchState({ node });
+  }
+
+  @Action(SetMapping)
+  setMapping({ patchState }: MedicalIllustrationContext, { url }: SetMapping): void {
+    parse(url || '', {
```

Comment:
When is it ever appropriate to send an empty string url to papaparse? Does the actually end up with an empty array as a result rather than an error because it thinks its the csv string to parse?

Reply:
It's not needed, I probably left this in by accident while working on it.